packages/amplify_api/example/lib/main.dart

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+import 'dart:async';
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'amplifyconfiguration.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatefulWidget {
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  String _result = '';
+  bool _isAmplifyConfigured = false;
+  Amplify amplify = new Amplify();
+
+  @override
+  void initState() {
+    super.initState();
+    initPlatformState();
+  }
+
+  Future<void> initPlatformState() async {
+    AmplifyAPI apiPlugin = AmplifyAPI();
+
+    await amplify.addPlugin(apiPlugins: [apiPlugin]);
+
+    // Configure
+    await amplify.configure(amplifyconfig);
+    setState(() {
+      _isAmplifyConfigured = true;
+    });
+  }
+
+  query() async {
+    String graphQLDocument = '''query MyQuery {
+      listBlogs {
+        items {
+          id
+          name
+          createdAt
+        }
+      }
+    }''';
+
+    var operation = await Amplify.API
+        .query<String>(request: GraphQLRequest(document: graphQLDocument));
```

Comment:
I didn't ask this in the design review, but last chance before it's permanent -- there's no way to omit this `<String>` syntax, right?

Reply:
It doesnt need to be provided explicitly as long as it can be inferred from somewhere, for example the `GraphQLRequest` thats passed in. Otherwise it will be dynamic. 
The documentation [here]( https://github.com/dart-lang/sdk/blob/master/pkg/dev_compiler/doc/GENERIC_METHODS.md#calling-generic-methods) provides more details.